File: modules/TaggedSentence.lua
require 'torch'
require 'nn'

local TaggedSentence = torch.class('nn.TaggedSentence')

-- A nn.TaggedSentence keeps.
function TaggedSentence:__init(words, tags)
  assert(#words == #tags, 'Word count must match tag count!')
  self.words = words
  self.tags = tags
end

function TaggedSentence:size()
  return #self.words
end

function TaggedSentence:__tostring__()
  local size = self:size()
  local result = ''
  for i = 1, size - 1 do
    result = result .. self.words[i] .. '_' .. self.tags[i] .. ' '
  end
  return result .. self.words[size] .. '_' .. self.tags[size]
end

File: theano_port/TaggedSentence.py

class TaggedSentence(object):
  
  def __init__(self, words, tags):
    if len(words) != len(tags):
      raise ValueError('Word count must match tag count!')
    self.words = words
    self.tags = tags

  def __len__(self):
    return len(self.words)

  def __str__(self):
    return ' '.join([self.words[i] + '_' + self.tags[i] for i in xrange(0, len(self.words))])
